docs/tutorials/getting-started/rust.mdx

```diff
@@ -0,0 +1,164 @@
+import FlagdContent from '@site/src/components/custom/tutorial/flagd-content.mdx';
+import FlagdChangeContent from '@site/src/components/custom/tutorial/flagd-change-content.mdx';
+
+# Getting Started with the OpenFeature Rust SDK and Axum
+
+## Introduction
+
+This tutorial explains how to integrate the OpenFeature Rust SDK into an Axum application using the flagd provider. In this guide you will learn how to:
+
+- Create a new Rust project with Cargo
+- Update dependencies to include Docker-based flagd configuration
+- Set up an Axum server with a custom AppState that holds a FlagdProvider
+- Evaluate a feature flag and return a custom message with diff markers to highlight changes
+
+## Requirements
+
+- Rust stable (1.80.2 or later recommended)
+- Cargo
+- Docker (to run the flagd service)
+- A valid flagd configuration file (e.g. `flags.flagd.json`)
+
+## Walk-through
+
+### Step 1: Create a new Rust project
+
+```bash
+cargo new axumstart
+```
+
+Then navigate to the project directory:
+
+```bash
+cd axumstart
+```
+
+### Step 2: Update Cargo.toml dependencies
+
+In your Cargo.toml add the following dependencies:
+
+```toml
+[dependencies]
+axum = "0.6"
+tokio = { version = "1", features = ["full"] }
+serde_json = "1.0"
+tracing = "0.1"
+open-feature = "^0"
+open-feature-flagd = "^0"
+```
+
+Adjust version numbers according to the latest releases.
+
+### Step 3: Set Up the Application
+
+Create or modify your `src/main.rs` with the following code. Notice the diff markers to indicate new additions:
+```rust
+// diff-remove-block-start
+fn main() {
+    println!("Hello, world!");
+}
+// diff-remove-block-end
+
+// diff-add-block-start
+use axum::{
+    extract::State,
+    http::StatusCode,
+    response::IntoResponse,
+    routing::get,
+    Json, Router,
+};
+use open_feature_flagd::{FlagdOptions, FlagdProvider};
+use open_feature::{EvaluationContext, provider::FeatureProvider};
+use serde_json::json;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct AppState {
+    pub flag_provider: Arc<FlagdProvider>,
+}
+
+async fn hello_handler(State(state): State<AppState>) -> impl IntoResponse {
+    // Create an evaluation context (custom fields can be added if required)
+    let context = EvaluationContext::default();
+
+    // Evaluate the "welcome-message" feature flag.
+    let flag_eval = state.flag_provider.resolve_bool_value("welcome-message", &context).await;
+    let message = match flag_eval {
+        Ok(result) if result.value => "Hello, welcome to this OpenFeature-enabled website!",
+        _ => "Hello!",
+    };
+
+    (StatusCode::OK, Json(json!({ "message": message })))
+}
+
+#[tokio::main]
+async fn main() {
+    // Initialize the flagd provider with default options.
+    let flag_provider = Arc::new(
+        FlagdProvider::new(FlagdOptions {
+            cache_settings: None,
+            ..Default::default()
+        })
+        .await
+        .expect("Failed to create FlagdProvider"),
```

Comment:
```suggestion
        .expect("Failed to connect to flagd. Please confirm flagd is running."),
```

Optional but it may make the error message easier to understand.